napari/layers/labels/labels.py

```diff
@@ -262,6 +266,24 @@ def n_dimensional(self, n_dimensional):
         self._n_dimensional = n_dimensional
         self.events.n_dimensional()
 
+    @property
+    def contour(self):
+        """bool: displays contours of labels instead of shaded regions."""
+        return self._contour
+
+    @contour.setter
+    def contour(self, contour, refresh=True):
+        self._contour = contour
+        self.events.contour()
+
+        if refresh is True:
+            self._save_history()
+
+        self._raw_to_displayed(self.data)
+
+        if refresh is True:
+            self.refresh()
```

Comment:
Why do you add the `refresh=True` and the `self._raw_to_displayed(self.data)` call?

Does just having `self.refresh()` at the end not always work? We do something similar here https://github.com/napari/napari/blob/2a6adfa4fdc58be96140b04774ae2c7ddc1701b0/napari/layers/labels/labels.py#L452-L455

Reply:
I also wouldn't include the `_save_history` call

Reply:
No particular reason, I guess I overlooked this part, but yes it works without the `self._raw_to_displayed(self.data)` call.
Also removed the `_save_history` call.